pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/client/SpoonDataServiceLocalClientConfigurer.java

```diff
@@ -81,9 +85,7 @@ private synchronized void setup( DataServiceClientService clientService ) {
 
       clientService.setRepository( spoon.getRepository() );
       clientService.setMetaStore( spoon.getMetaStore() );
-      ThinConnection.localClient = clientService;
-    } else {
-      ThinConnection.localClient = null;
     }
+    // else, should we clean (set to null) the clientService repository and metastore?
```

Comment:
Scanning through the code it doesn't appear those values are even used any more.  Both the .getRepository and .getMetastore methods are marked deprecated in RemoteClient and appear to be no-ops in the implementations.  I think "locator" classes are used to retrieve metastore and repos now.

So yeah, you might be able to clean this out more and eliminate some unnecessary code.  If it looks too risky or uncertain you could leave as is and add a comment to revisit.